### DISCUSSION TITLE
Atom's structure

### DISCUSSION THREAD

**Discussion by iorran** [pmndrs/jotai#1087] [General]
Guys,

I need to create a pattern to be reused in my company. 

I have some questions regarding code organization, which I would like to discuss.

1. Do you guys place your atoms in a separate file?
2. If you have an object **Person { id: number, name: string }**, what are the differences between:
``
const idAtom = atom()
const nameAtom = atom() 
``
vs

``
const personAtom = atom({ id: undefined, name: undefined }) 
``
vs

``
const personAtom = atom({ id: atom(undefined), name: atom(undefined) }) 
``
3. Do you guys export only derived atoms ?
4. Share how do you guys organize it, please!

**Comment by TwistedMinda** [comment]
@iorran As for the app structure, I was wondering too and tried to tackle the problem [in this discussion](https://github.com/pmndrs/jotai/discussions/1172). Give it a check when you have time.

As for the difference between your 2 examples. It's basically 2 atoms vs 1.
- If you update `personAtom`, all components listening to this atom will re-render.
- If you had split them to `idAtom` and `nameAtom`, you could update them separately and hence have more control on what parts of your app should re-render.

Of course, you could reach the same result with 1 `personAtom` and 2 derived atoms (1 for id, 1 for name) ;)
Depends on what you need/feel like.

Remember jotai is un-opinionated and let's you be creative on how you use it.
Have fun
